app/scripts/controllers/paymentModal.js

```diff
@@ -39,6 +39,13 @@ angular.module('confRegistrationWebApp')
         alert('Transaction amount must be a positive number.');
         return;
       }
+      if ($scope.newTransaction.paymentType === 'CREDIT_CARD') {
+        var validationError = ccp.validateCardNumber($scope.newTransaction.creditCard.number || '');
+        if(validationError){
+          alert('Please enter a valid card number. The ' + validationError + '.');
+          return;
+        }
```

Comment:
payment.js is for when the user is submitting a payment, the paymentModal.js is for when an admin is manually entering a payment.